### PR TITLE
Modify Trade Republic PDF-Importer to support new transaction

### PR DESCRIPTION
**Pull request by Nirus2000**
https://forum.portfolio-performance.info/t/pdf-import-von-trade-republic/5107/508 
https://forum.portfolio-performance.info/t/pdf-import-von-trade-republic/5107/510

### CODE REVIEW

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java [hunk 1]
```diff
@@ -1155,6 +1156,246 @@ public void testKontoauszug05()
...
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(0L));
+        assertThat(countBuySell(results), is(0L));
+        assertThat(countAccountTransactions(results), is(20L));
+        assertThat(results.size(), is(20));
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+
+        // assert transaction
+        assertThat(results, hasItem(interest(hasDate("2024-04-01"), hasAmount("EUR", 147.34), //
```
Comment (christen90) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1177]: I would mark this as NonImportable. We don't know the taxes and fee situation.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java [hunk 2]
```diff
@@ -1155,6 +1156,246 @@ public void testKontoauszug05()
...
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(0L));
+        assertThat(countBuySell(results), is(0L));
+        assertThat(countAccountTransactions(results), is(35L));
+        assertThat(results.size(), is(35));
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+
+        // assert transaction
+        assertThat(results, hasItem(interest(hasDate("2024-04-01"), hasAmount("EUR", 53.89), //
```
Comment (christen90) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1275]: also nonImportable

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java (hunk 1 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1177]: Why? Are they not interest payments? (Interest on the assets?)

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java [hunk 3]
```diff
@@ -1155,6 +1156,246 @@ public void testKontoauszug05()
...
+                        hasSource("Kontoauszug07.txt"), hasNote("TooGoodT"))));
+
+        // assert transaction
+        assertThat(results, hasItem(removal(hasDate("2024-04-14"), hasAmount("EUR", 39.86), //
+                        hasSource("Kontoauszug07.txt"), hasNote("Lidl sagt Danke"))));
+
+        // assert transaction
+        assertThat(results, hasItem(deposit(hasDate("2024-04-14"), hasAmount("EUR", 0.08), //
+                        hasSource("Kontoauszug07.txt"), hasNote("Visa Geld zurueck Aktion"))));
+
```
Comment (christen90) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1305]: missing
16.4.2024 18,34 ALIEXPRESS.COM
16.4.2024 17,53 ALIEXPRESS.COM

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java [hunk 4]
```diff
@@ -1155,6 +1156,246 @@ public void testKontoauszug05()
...
+        // assert transaction
+        assertThat(results, hasItem(removal(hasDate("2024-04-17"), hasAmount("EUR", 18.34), //
```
Comment (christen90) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1307]: Saldo gets increased --> deposit

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java [hunk 5]
```diff
@@ -1304,6 +1322,173 @@ private void addAccountStatementTransaction()
...
+        depositBlock.set(new Transaction<AccountTransaction>()
+
+                        .subject(() -> {
+                            AccountTransaction accountTransaction = new AccountTransaction();
+                            accountTransaction.setType(AccountTransaction.Type.DEPOSIT);
+                            return accountTransaction;
+                        })
+
+                        .section("date", "amount", "currency") //
+                        .documentContext("year") //
```
Comment (christen90) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1353]: what if user generates statement from Dec 2023 to Jan 2024?

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java [hunk 6]
```diff
@@ -1304,6 +1322,173 @@ private void addAccountStatementTransaction()
...
+                                        section -> section //
+                                                        .attributes("date", "note", "amount", "currency") //
+                                                        .documentContext("year") //
+                                                        .match("^(?<date>[\\d]{2} [\\w]{3,4}([\\.]{1})?)[\\s]$")
+                                                        .match("^[\\d]{4} Kartentransaktion (?<note>.*) (?<amount>[\\.,\\d]+) (?<currency>\\p{Sc}) [\\.,\\d]+ \\p{Sc}$") //
+                                                        .assign((t, v) -> {
+                                                            // @formatter:off
+                                                            // Is type is "Visa Geld zurueck Aktion" change from REMOVAL to DEPOSIT
+                                                            // @formatter:on
+                                                            if ("Visa Geld zurueck Aktion".equals(v.get("note")))
```
Comment (christen90) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1390]: how to catch other paybacks, refunds?

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java [hunk 7]
```diff
@@ -1304,6 +1322,173 @@ private void addAccountStatementTransaction()
...
+
+                        .wrap(t -> {
+                            if (t.getCurrencyCode() != null && t.getAmount() != 0)
+                                return new TransactionItem(t);
+                            return null;
+                        }));
+
+        // @formatter:off
+        // 03 Apr.
+        // 024 Gebühren Trade Republic Card 5,00 € 49.997,41 €
```
Comment (christen90) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1431]: **2**024

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java [hunk 8]
```diff
@@ -1304,6 +1322,173 @@ private void addAccountStatementTransaction()
...
+                                        // @formatter:on
+                                        .section("year") //
+                                        .match("^Erstellt am [\\d]{2} .* (?<year>[\\d]{4})$") //
+                                        .assign((ctx, v) -> ctx.put("year", v.get("year"))));
+
+        this.addDocumentTyp(type);
+
+        // @formatter:off
+        // 02 Apr. Überweisung Einzahlung akzeptiert: DE7243872432 auf 2024 DE7243872432 1.200,00 € 51.352,41 €
+        // @formatter:on
```
Comment (christen90) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1340]: other format:
30 Apr. 
2024 Überweisung
Einzahlung akzeptiert: DE00000000000000000000 auf 
DE00000000000000000000 1.200,00 € 18.514,32 €

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java (hunk 1 above)
Comment (christen90) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1177]: Yes, but we don't know if taxes were applied and manipulate the payed taxes overview. I could confirm this in Dec 2023. Freibetrag was exhausted and taxes were applied. No hint about taxes was visible in the export.

Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1177]: We need a pdf debug to check this

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java (hunk 5 above)
Comment (Nirus2000) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1353]: We need a pdf debug to check this.
Did you have anyone?!

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java (hunk 6 above)
Comment (Nirus2000) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1390]: We need a pdf debug to check this.
Did you have anyone?!

Comment (christen90) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TradeRepublicPDFExtractor.java:1390]: See aliexpress booking on 17th Apr. Saldo gets increased compared to previous row.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java (hunk 1 above)
Comment (christen90) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/traderepublic/TradeRepublicPDFExtractorTest.java:1177]: I can't provide a debug of something which is not contained. You can see in both exports an interest booking. Both do not contain taxes.

**Pull request merged**